src/Runtime/SonataPageRuntime.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\PageBundle\Runtime;
+
+use Sonata\PageBundle\Request\RequestFactory;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner;
+use Symfony\Component\Runtime\RunnerInterface;
+use Symfony\Component\Runtime\SymfonyRuntime;
+
+class SonataPageRuntime extends SymfonyRuntime
+{
+    private string $multisite;
+
+    /**
+     * @param  array{
+     *      debug?: bool|null,
+     *      env?: string|null,
+     *      disable_dotenv?: bool|null,
+     *      project_dir?: string|null,
+     *      prod_envs?: string[]|null,
+     *      dotenv_path?: string|null,
+     *      test_envs?: string[]|null,
+     *      use_putenv?: bool|null,
+     *      runtimes?: array<mixed>|null,
+     *      error_handler?: string|false,
+     *      env_var_name?: string,
+     *      debug_var_name?: string,
+     *      dotenv_overload?: bool|null,
+     *      multisite?: string,
+     *    }  $options  $options
+     */
+    public function __construct(array $options = [])
+    {
+        if (!isset($options['multisite'])) {
+            throw new \InvalidArgumentException('The option multisite is required');
+        }
+
+        $this->multisite = $options['multisite'];
+        unset($options['multisite']);
+        parent::__construct($options);
+    }
+
+    public function getRunner(?object $application): RunnerInterface
```

Comment:
Ok as I supected run time works with this code ;)

```php
<?php

declare(strict_types=1);

/*
 * This file is part of the Sonata Project package.
 *
 * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace Sonata\PageBundle\Runtime;

use Sonata\PageBundle\Request\SiteRequest;
use Symfony\Component\HttpKernel\HttpKernelInterface;
use Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner;
use Symfony\Component\Runtime\RunnerInterface;
use Symfony\Component\Runtime\SymfonyRuntime;

class SonataPageRuntime extends SymfonyRuntime
{
    public function getRunner(?object $application): RunnerInterface
    {
        if (!$application instanceof HttpKernelInterface) {
            return parent::getRunner($application);
        }

        return new HttpKernelRunner($application, SiteRequest::createFromGlobals());

    }
}
```

and you don't need to pass `multisite` on composer.json

Reply:
and I just search and `RequestFactory` is not used anywhere, then we can deprecate the whole class!

Reply:
well if you want you can keep the local like this


```php
        if ($application instanceof HttpKernelInterface) {
            return new HttpKernelRunner($application, SiteRequest::createFromGlobals());
        }
        
        return parent::getRunner($application);
```

it is up to you ;)

Reply:
`and you don't need to pass multisite on composer.json`

You are right, you could get rid of this, but as the Sonata\PageBundle\Request\RequestFactory line 60 it prevents the creation of the Request::setFactory twice as the first one is already done. So in my point of view, no, you should keep it. 

`well if you want you can keep the local like this`
What do you mean ?

Reply:
Do you have a branch i could use to try if i's ok in my set up ?

Reply:
> You are right, you could get rid of this, but as the Sonata\PageBundle\Request\RequestFactory line 60 it prevents the creation of the Request::setFactory twice as the first one is already done. So in my point of view, no, you should keep it.

It was there because it was used in front controller and we have only one, as it was there we need to to check what was passing, because it could be `host` or `host_path`, but as we are using runtime, it will use host_path... only.

then this if gonna be always true! it is not needed.

> What do you mean ?

Sorry it was a typo, I edited the comment.

> Do you have a branch i could use to try if i's ok in my set up ?

nope, I just used your branch and update directly on vendor folder, if you copy and past the code above into your runtime, you gonna see the same behaviour then me ;)

Reply:
> Do you have a branch i could use to try if i's ok in my set up ?

or do you think is necessary I create a branch with this?

Reply:
That's right, you can go directly to `SiteRequest::createFromGlobals()`  it could simplify the all process. kudo!

As `RequestFactory::create` and   `RequestFactory::createFromGlobals` seams useless in the case that only 
`host_with_path` ad `host_with_path_by_locale` should be used with the Runtime. 

But, how to prevent user to use multisite:host params with the runtime ? Would it be any impact ?

Reply:
> But, how to prevent user to use multisite:host params with the runtime ? Would it be any impact ?

it is used in others class via param, I saw that it was used as a event listener, and we are also duplicating some logical there, I believe we can also simplify this, But I didn't start this topic to not complicate this PR.

But I guess it will be easier to handle it later since it is a internal logical.

btw, do not forgot to add final class for your runtime.

Reply:
> But, how to prevent user to use multisite:host params with the runtime

Note1: we still need to keep this param because it is used in some event listeners, But for runtime, it is not important ;)

Reply:
@GeraudBourdin when you update let me know then I can review again 😃

Reply:
Sorry for the delay, i had work to do. 
So PR updated with the new SiteRequest call, updated tests and doc. 
Let me know what' could be improved.